moccasin/config.py

```diff
@@ -122,9 +127,79 @@ def _set_boa_env_and_db(self) -> _AnyEnv:
                 )
             self.chain_id = expected_chain_id if not self.chain_id else self.chain_id
 
+            # 5. Set the explorer type and uri if not set by default ones
+            if self.chain_id is not None and (
+                self.explorer_uri is None or self.explorer_uri is None
+            ):
+                # Review: Is this down here even good?
+                from moccasin.constants.vars import DEFAULT_NETWORKS_BY_CHAIN_ID
+
+                default_network_info = DEFAULT_NETWORKS_BY_CHAIN_ID.get(
+                    self.chain_id, {}
+                )
+                self.explorer_uri = (
+                    self.explorer_uri
+                    if self.explorer_uri is not None
+                    else default_network_info.get("explorer", None)
+                )
+                self.explorer_type = (
+                    self.explorer_type
+                    if self.explorer_type is not None
+                    else default_network_info.get("explorer_type", None)
+                )
+
         boa.env.nickname = self.name
         return boa.env
 
+    def moccasin_verify(
+        self, contract: VyperContract | ZksyncContract
+    ) -> "VerificationResult":
+        """Verifies a contract using your moccasin.toml config."""
+        verifier_class = self.get_verifier_class()
+        verifier_instance = verifier_class(self.explorer_uri, self.explorer_api_key)
+        if self.is_zksync:
+            import boa_zksync
+
+            with boa.set_verifier(verifier_instance):
+                result = boa_zksync.verify(contract, verifier_instance)
+                return result
+        else:
+            with boa.set_verifier(verifier_instance):
+                result = boa.verify(contract, verifier_instance)
+                return result
```

Comment:
No need for boa.set_verifier if we are passing the instance

```suggestion
            return boa_zksync.verify(contract, verifier_instance)
        return boa.verify(contract, verifier_instance)
```

The only reason why we need a separate method for boa_zksync is that we miss the `constructor_calldata` in `boa.verify`. That's fixed in https://github.com/vyperlang/titanoboa/pull/330